core/widgets/SubmitButtonWidget.php

```diff
@@ -53,21 +53,28 @@ public function init()
     }
 
     /**
-     * {@inheritDoc}
+     * Add beforeSubmit handler which disables button and replaces button text
+     * @return string
      */
     public function run()
     {
-        $js = [
-            'this.disabled=true;',
-        ];
 
-        if ($this->pushed) {
-            $js[] = "this.innerHTML='{$this->pushed}';";
-        }
+        $this->view->registerJs("
+            $(document).on('beforeSubmit', function (e) {                
+                var buttonSelector = $(e.target).find(':submit');
+                var formSelector = $(e.target);
+                if (formSelector.find('div.has-error').length === 0) {
```

Comment:
i assume `has-error` is a class, what if someone changes this class? Is it configurable in activeform/activefield?

Reply:
Good catch! I'll add an optional variable

Reply:
The variable is not needed if we have active form context (if the error class is configured in this object)